src/utils/userLocation.ts

```diff
@@ -55,26 +86,32 @@ export const enterExternalRoom = ({ userId, room }: EnterExternalRoomProps) => {
 };
 
 export const useUpdateTimespentPeriodically = (
-  user: UserInfo | undefined,
-  roomName: string
+  roomName: string,
+  userId?: string
 ) => {
-  const shouldUseInterval = user && roomName;
+  const shouldUseInterval = userId && roomName;
 
   useInterval(
     () => {
-      // Time spent is currently counted multiple time if multiple tabs are open
-      if (!user || !roomName) return;
+      // @debt time spent is currently counted multiple time if multiple tabs are open
+      if (!userId || !roomName) return;
 
       const firestore = firebase.firestore();
-      const doc = `users/${user.uid}/visits/${roomName}`;
-      const increment = firebase.firestore.FieldValue.increment(
-        LOCATION_INCREMENT_SECONDS
-      );
+      const doc = `users/${userId}/visits/${roomName}`;
```

Comment:
I still feel like we would be better to use the more verbose method of `firestore.doc(doc).collection('users').doc(userId)...etc` instead of this path, as I believe as it currently is, this would potentially allow a SQL injection/path traversal sort of issue.

Reply:
fixed in
https://github.com/sparkletown/sparkle/pull/1129/commits/53f57c40e24983470b7552d4ebed6f1715ff74ba